### PR TITLE
Dbus-[debug, AsRefClosure] + use cargo version + initial test file

### PR DESCRIPTION
**Pull request by Stunkymonkey**

### CODE REVIEW

File: durst/Cargo.toml [hunk 1]
```diff
@@ -3,6 +3,14 @@ name = "durst"
 version = "0.1.0"
 authors = ["Felix Buehler <account@buehler.rocks>"]
 edition = "2018"
+license = "GPL-3.0-or-later"
+repository = "https://github.com/durst-notification/durst"
+description = "dunst written Rust"
```
Comment (bebehei) [inline durst/Cargo.toml]: "dunst written in Rust" (I've removed the typo in the repo description :wink: )

File: durst/Cargo.toml [hunk 2]
```diff
@@ -3,6 +3,14 @@ name = "durst"
 version = "0.1.0"
 authors = ["Felix Buehler <account@buehler.rocks>"]
 edition = "2018"
+license = "GPL-3.0-or-later"
```
Comment (bebehei) [inline durst/Cargo.toml]: We should rather add the GPL as a license file, which is rather common. (can be done later, too).

File: durst/src/main.rs [hunk 3]
```diff
@@ -43,9 +43,9 @@ impl interface::OrgFreedesktopNotifications for Notifications {
     fn get_server_information(&self) -> Result<(String, String, String, String), Err> {
         println!("running getserverinformation");
         Ok((
-            "durst".to_string(),
+            env!("CARGO_PKG_NAME").to_string(),
```
Comment (bebehei) [inline durst/src/main.rs:59]: At the end of the day, the user will run durst without cargo. The env variable won't be there anymore.

File: durst/src/main.rs [hunk 4]
```diff
@@ -12,7 +13,7 @@ type Err = tree::MethodErr;
 
 impl interface::OrgFreedesktopNotifications for Notifications {
     fn get_capabilities(&self) -> Result<Vec<String>, Err> {
-        println!("get_capabilities");
+        debug!("get_capabilities");
```
Comment (bebehei) [inline durst/src/main.rs:24]: Nice! :+1:

File: durst/src/main.rs (hunk 3 above)
Comment (Stunkymonkey) [reply durst/src/main.rs:59]: i think this will be resolved during the compilation

Comment (Stunkymonkey) [reply durst/src/main.rs:59]: https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

**Review by bebehei — APPROVED** [review]

**Pull request merged**